### ISSUE TITLE
Portal help (When leaving Nether get sent to Overworld-2 not Overworld-1) 

### ISSUE THREAD

**Issue by LluviaWolf** [SpongePowered/SpongeForge#3229]
**I am currently running**

- SpongeForge version: spongeforge-1.12.2-2838-7.2.3.jar
- Forge version: forge-14.23.5.2854
- Java version: 8 
- Operating System: MacOS  

- Plugins/Mods:
-Pixelmon Reforged
-Journey Map's 
-Armourer's Workshop 
-Pixelmon OST 
-Pixelmon Extras 
-Gameshark
-BOP Patch
-BiomesOPlenty 
-SimpleDexRewards 
-Even More Pixelmon Commands 
-Pixelmon Broadcasts 
-Pixelmon Auction 
-Pixel Money 
-Pixel Event 
-Rustles
-Move Skill Controller 
-Trainer Commands 
-Tournaments 
-Wonder Trade 
-Pixel Hunt 
-Pixelmon Economy Bridge 
-Vanilla Builders Extension 
-Chisel 
-Chisel and Bits 
-Tinkers Construct 
-Bewitchment
-Spartan Shield 
-Spongeforge 
-BuycraftX 
-CatClearLag 
-MagiBridge 
-Nucleus 
-LuckPerms 
-Worldedit 
-Griefprevention 
-Holograms 
-Total Economy 
-Carrot Shop 
-HuskyCrates
-AC Negativity 
-Packet Gate
-GTS
-Impactor API
-Heads Evolved
-TeslaPowered
-FlashSpec-2.3.1.jar

**Issue Description**
Our main world is called **world** but since it did not feature specific things (BOP) we used Nucleus to create a "new world" and we called it **Blue**.
But now every time we go to the Nether in **Blue** we get send to the Nether but when we leave the Nether it sends us to **world** instead of **Blue** and the same goes for the End Portal.

How do we change that ?

maybe the picture can help. This is inside our server when we open our folder called **world**
<img width="485" alt="Schermafbeelding 2020-11-22 om 23 34 48" src="https://user-images.githubusercontent.com/71126315/99919116-af19c280-2d1b-11eb-9a88-17bd48b8fb10.png">

Basically we want it so that when players leave the Nether **DIM-1** to get send back to **Blue** NOT **world**
and the same thing for the End **DIM1**

Hope you guys can help us out.

**Comment by Zidane** [comment]
Take a look at portal-agents in your world config file for "Blue"

**Comment by LluviaWolf** [comment]
dumb question but where is that located ?

**Comment by LluviaWolf** [comment]
for some reason I can not seem to find the world config file.
<img width="296" alt="Schermafbeelding 2020-11-23 om 00 01 54" src="https://user-images.githubusercontent.com/71126315/99919678-213fd680-2d1f-11eb-80db-f55db2b614e1.png">
I checked the folder called Config and couldn't find anything... is it in the world folder ?

**Comment by LluviaWolf** [comment]
<img width="715" alt="Schermafbeelding 2020-11-23 om 00 03 59" src="https://user-images.githubusercontent.com/71126315/99919724-6a902600-2d1f-11eb-9782-066c80d6a213.png">
this is inside the folder called Blue

**Comment by thibaulthenry** [comment]
@LluviaWolf 

You can't have a nether world that redirect to two different overworlds

**Comment by LluviaWolf** [comment]
I know that. I only want it to connect with the overworld "Blue" not "world"

But I can't find where the portal agent code is located. In which config file is it, where do I find it ?

**Comment by dualspiral** [comment]
> Take a look at portal-agents in your world config file for "Blue"

You actually want the `DIM-1` config file. I think you want `/config/sponge/worlds/minecraft/nether/DIM-1/world.conf`. Then paste the following:

```
sponge {
    world {
        # A list of all detected portal agents used in this world.
        # In order to override, change the target world name to any other valid world.
        # Note: If world is not found, it will fallback to default.
        portal-agents {
            "minecraft:default_nether"="Blue"
            "minecraft:default_the_end"="Blue"
        }
    }
}
```

Same for DIM1, in `/config/sponge/worlds/minecraft/the_end/DIM1/world.conf`.

**Comment by dualspiral** [comment]
Also, for note, this isn't a bug in Sponge, this is a request for support. As there is nothing we need to do I'm closing this issue (but feel free to reply to this issue, I just don't want it left open when there is nothing we need to do.)

---

**Github issues are for reporting bugs and requesting features in Sponge.** As this issue is a request for support, I am closing this issue.

The following resources are available to support you in using Sponge - please use these in future unless you are reporting a bug or asking for a new feature:

* Sponge Docs (available in multiple languages): https://docs.spongepowered.org
* Sponge Forums: https://forums.spongepowered.org/
* Sponge Discord: https://discord.gg/sponge

**Comment by thibaulthenry** [comment]
I think you forgot the world node @dualspiral 
I would say 

```
sponge {
        world {
                # A list of all detected portal agents used in this world.
                # In order to override, change the target world name to any other valid world.
                # Note: If world is not found, it will fallback to default.
                portal-agents {
                    "minecraft:default_nether"="Blue"
                    "minecraft:default_the_end"="Blue"
                }
        }
}
```

**Comment by dualspiral** [comment]
Yes, you're correct. @LluviaWolf, you'll want what @thibaulthenry suggested.

**Comment by LluviaWolf** [comment]
got it like this 

```
# 1.0
# 
# # If you need help with the configuration or have any questions related to Sponge,
# # join us at the IRC or drop by our forums and leave a post.
# 
# # IRC: #sponge @ irc.esper.net ( https://webchat.esper.net/?channel=sponge )
# # Forums: https://forums.spongepowered.org/
# 

sponge {
    # Configuration options that will affect all worlds.
    world {
        # If 'true', this world will load on startup.
        load-on-startup=true
        # A list of all detected portal agents used in this world.
        # In order to override, change the target world name to any other valid world.
        # Note: If world is not found, it will fallback to default.
        portal-agents {
                "minecraft:default_nether"="Blue"
                "minecraft:default_the_end"="Blue"
                }
        }
}

```
But it doesn't work.

**Comment by thibaulthenry** [comment]
Oups, remove the quotes from Blue.

```
portal-agents {
        "minecraft:default_nether"=Blue
        "minecraft:default_the_end"=Blue
}
```

**Comment by LluviaWolf** [comment]
really weird I did that and it still sends me to `world` instead of `Blue`

**Comment by thibaulthenry** [comment]
Well, I think dual and me messed up.

In `/config/sponge/worlds/minecraft/nether/DIM-1/world.conf` (your default nether). You want this nether to redirect to Blue overworld.
Add

```
sponge {
        world {
                portal-agents {
                        "minecraft:default_overworld"=Blue
                }
        }
}
```

In `/config/sponge/worlds/minecraft/the_end/DIM1/world.conf` (your default end). You want this end to redirect to Blue overworld.
Add 

```
sponge {
        world {
                portal-agents {
                        "minecraft:default_overworld"=Blue
                }
        }
}
```

In `/config/sponge/worlds/minecraft/overworld/Blue/world.conf` (your overworld Blue), you have nothing to do **BECAUSE** you are targetting default nether and default end, which are the default redirections for all your overworlds (see at the end of `/config/sponge/global.conf`).

**In the case** would have another Nether or End worlds (whatever their names), you would have to do 

`/config/sponge/worlds/minecraft/overworld/<whatever-overworld-name>/world.conf`

```
sponge {
        world {
                portal-agents {
                        "minecraft:default_the_nether"=<whatever-nether-name>
                        "minecraft:default_the_end"=<whatever-end-name>
                }
        }
}
```

Sorry for the inconvenience

**Comment by LluviaWolf** [comment]
<img width="768" alt="Schermafbeelding 2020-11-24 om 11 17 57" src="https://user-images.githubusercontent.com/71126315/100080973-c6a19a00-2e46-11eb-9f95-d2ca18ed4b46.png">

I have it like this in `/config/sponge/worlds/minecraft/nether/DIM-1/world.conf` 
and it still sends me to `world`
This is getting really frustrating :<

**Comment by LluviaWolf** [comment]
`Blue` is not my default overworld though... `world` is my default overworld could that be a problem ?

**Comment by thibaulthenry** [comment]
No, the "minecraft:default_overworld" is just the type of the portal-agent.

Can you come to the Sponge Discord so it's easier to talk ?

**Comment by LluviaWolf** [comment]
Sure I'm in the Sponge Discord server feel free to dm me there Lluvia#7150

**Comment by thibaulthenry** [comment]
I would like to end up this issue to warn people which would try to setup portal agents.
Do not forget to stop the server before editing the worlds configuration files, and then restart the server once you ended up modifications.

If you edit the files while the server is running, the current configurations loaded in memory on the server will overwrite any change you made.
